### PR TITLE
Feature/sc 5797 dashboard regular homeworks are shown for a teacher

### PR DESCRIPTION
**Pull request by alstanki**
# Description
New test : showRegularTaskOnDashboard created

## Links to Tickets or other pull requests
https://ticketsystem.hpi-schul-cloud.org/browse/SC-5797

## Changes
- created new test : showRegularTaskOnDashboard
-added new methods, step-definitions,

## Approval for review
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.hpi-schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: page-objects/pages/DashboardPage.js [hunk 1]
```diff
@@ -3,69 +3,187 @@
...
+	return expect(isTaskVisible, msg + resultMsg).to.equal(true)
+}
 
-	expectedValue
-		? expect(elementOfTask, msg).to.be.true
-		: expect(elementOfTask, msg).to.be.false;
+async function isElementOnTaskVisible(elementName, taskName, section, expectedValue) {
+	const task = await getTaskWithNameInSection(taskName, section);
+	const elementSel = getTaskElementSel(elementName);
+	const isTaskWithElement = await (await task.element.$(elementSel)).isDisplayed();
```
Comment (haniapalac) [inline page-objects/pages/DashboardPage.js:171]: same here - double await

File: page-objects/pages/DashboardPage.js [hunk 2]
```diff
@@ -3,69 +3,187 @@
...
-	taskCompleted: 'span[data-testid="task-submitted"]',
-	taskGraded: 'span[data-testid="task-graded"]',
-	dashboardTitleList: 'a[data-testid="dashboard-tasks-title"]',
+	deadline: 'span[data-testid="task-due-date"]',
+	submitted: 'span[data-testid="task-submitted"]',
+	graded: 'span[data-testid="task-graded"]',
+};
+
+const taskSection = {
+	assigned: 'section[data-testid="assigned-tasks-section"]',
```
[the remark this answers is not in the document]
Comment (alstanki) [reply page-objects/pages/DashboardPage.js]: ok. I changed it to "published"

File: page-objects/pages/DashboardPage.js [hunk 3]
```diff
@@ -3,69 +3,187 @@
...
+		getComparisonOfNumberOfTasks(await element.$(taskElement.submitted))
+	);
+	const comparisonOfNumberOfGradedTasks = await tryGetValue(async () =>
+		getComparisonOfNumberOfTasks(await element.$(taskElement.graded))
+	);
+
+	return {
+		element: element,
+		name: await tryGetValue(async () => (await element.$(taskElement.name)).getText()),
+		courseName: await tryGetValue(async () => await (await element.$(taskElement.courseName)).getText()),
```
[the remark this answers is not in the document]
Comment (alstanki) [reply page-objects/pages/DashboardPage.js]: $ - needs await and getText as well

File: page-objects/pages/DashboardPage.js (hunk 1 above)
Comment (alstanki) [reply page-objects/pages/DashboardPage.js]: $ - needs await and isDisplayed as well

File: page-objects/pages/DashboardPage.js [hunk 4]
```diff
@@ -3,69 +3,187 @@
...
+
+	return {
+		current: result[1],
+		expected: result[2],
+	};
+}
+
+async function tryGetValue(func) {
+	try {
+		let val = await func();
```
[the remark this answers is not in the document]
Comment (alstanki) [reply page-objects/pages/DashboardPage.js]: I changed a bit the implemention of the method that used this one. This method is no longer necessary. I removed it.

**Review by LukasSielski — APPROVED** [review]

**Review by haniapalac — DISMISSED** [review]

**Review by Grzesiek80 — APPROVED** [review]

**Pull request merged**